haystack/document_store/elasticsearch.py

```diff
@@ -699,10 +699,36 @@ def add_eval_data(self, filename: str, doc_index: str = "eval_document", label_i
         :type label_index: str
         """
 
-        docs, labels = eval_data_from_file(filename)
+        docs, labels = eval_data_from_json(filename)
         self.write_documents(docs, index=doc_index)
         self.write_labels(labels, index=label_index)
 
+    def add_eval_data_batchwise(self, filename: str, doc_index: str = "eval_document",
```

Comment:
I think the functionality of batch wise indexing is important. However, I'd suggest a different design. How about: 
1. Add a param `batch_size` to `add_eval_data()` instead of the new "add_eval_batchwise()"
2. Forward to the param `batch_size` from `add_eval_data()` to `write_documents()` and `write_labels()`
3. Implement the actual batching in `write_documents()` and `write_labels()` (e.g. here: https://github.com/deepset-ai/haystack/blob/281f9ff9709342bebd0f5b0e47bbab5a266b89e1/haystack/document_store/elasticsearch.py#L269)

With this design we can use the batching also for other calls of "write_documents()" and not only for eval data. It's also a bit cleaner to have a param instead of a separate "add_eval_data_batchwise()". What do you think?

If we agree on this, we should then do this change for all document stores.

Reply:
If we pass the generator extracting docs and labels from jsonl formatted squad files, we would need to iterate twice over docs, once in` write_documents()` and once in `write_labels()`. Not sure for now how to solve this...